### PR TITLE
Stats Widget: Fix dashboard styles impacted by widget styling

### PR DESCRIPTION
**Pull request by dognose24**
Related to #75360

## Proposed Changes

* Extract modified layout styles only for the Odyssey Stats page to an independent file.
* Use minimal base styles built from Calypso to reduce the impact on base elements, e.g., `<p>` and `<ul>`.

|Before|After|
|-|-|
|截圖 2023-04-18 下午9 42 05|截圖 2023-04-18 下午9 41 12|

## Testing Instructions

* Open your local Jetpack site.
* Check out the branch of the Jetpack to the latest `trunk`.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Navigate to `/wp-admin/index.php?odyssey_widget=1`.
* Ensure the `Dashboard` page styling works the same as the page without the feature flag `odyssey_widget=1`.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by kangzj — APPROVED** [review]
Thanks for fixing this. It works well for me! 👍

**Review by jsnmoon — APPROVED** [review]
Works well! Let's use this PR as an opportunity for a design review 🙂

**Review by jsnmoon — COMMENTED** [review]
Since we're expanding the scope of this PR to include designer reviews, I wanted to point out a two things:

image

1. The current top posts/pages & referrers are not linked. Could we make these individual top posts and referrers deeply link into Odyssey Stats?

image

2. The text in the Akismet module overflows to two lines. This likely needs a design solution.

**Comment by kangzj** [comment]
I did notice one thing that'll need change - in our design [p1HpG7-l7H-p2], it shows at most **5 items** for posts and referrers.

image

We are currently showing all (?), which doesn't look nice when there are many of them.

**Comment by kangzj** [comment]
Another thing is we'll need to format the numbers. I don't see any code handling that.

image

image

**Comment by kangzj** [comment]
I'm happy that we address these feedback in separate PRs.

Hey @filipanoscampos feel free to comment here and we'll follow up. Thanks 🙇

**Review by keoshi — APPROVED** [review]
LGTM!

**Comment by ederrengifo** [comment]
@jsnmoon @kangzj regarding the design issues:

1. Is this related to what we were exploring in the P2? To make these links point to their details instead of the posts?
2. Is there a way to force both cards to go to two lines simultaneously? Maybe some breakpoint?
Screen Shot 2023-04-19 at 09 48 46
3. I agree we should limit the posts and referrers to only 5
4. Do you need anything from the design regarding the numbers formating? Or is this a development issue to solve?

**Review by ederrengifo — COMMENTED** [review]
LGTM!

**Review by ederrengifo — APPROVED** [review]

**Comment by jsnmoon** [comment]
> 1. Is this related to what we were exploring in the P2? To make these links point to their details instead of the posts?

Yes! That's exactly what I was thinking.

> 2. Is there a way to force both cards to go to two lines simultaneously? Maybe some breakpoint?

This is possible, but I'll let @dognose24 make the final call for this.

> 3. I agree we should limit the posts and referrers to only 5

👍 

> 4. Do you need anything from the design regarding the numbers formating? Or is this a development issue to solve?

I don't think so! Definitely a development issue.

-----

@dognose24: Feel free to address these issues in this PR or in a follow-up PR.

**Comment by dognose24** [comment]
@kangzj @jsnmoon @keoshi @ederrengifo Thanks for all your reviews and feedback! Since this PR was expanded to use the final design review, I would prefer to fix all significant issues here.

> The current top posts/pages & referrers are not linked. Could we make these individual top posts and referrers deeply link into Odyssey Stats?

Updated in [31b00a8](https://github.com/Automattic/wp-calypso/pull/75897/commits/31b00a8460bd09bb841fb2dff8cc27a28316b99b).

> it shows at most 5 items for posts and referrers.

Fixed in [bf34c12](https://github.com/Automattic/wp-calypso/pull/75897/commits/bf34c126ad62839c9eec983267217fc64a513e4f).

> Another thing is we'll need to format the numbers.

Fixed in [4e43898](https://github.com/Automattic/wp-calypso/pull/75897/commits/4e438982c156254210e175d75135568bdfb27110), and the format for the module card numbers [is there already](https://github.com/Automattic/wp-calypso/blob/update/stats_widget_impacted_styles/apps/odyssey-stats/src/widget/modules.jsx#L15).

> Is there a way to force both cards to go to two lines simultaneously? Maybe some breakpoint?

Since the whole dashboard layout would change on different breakpoints, the window breakpoint may not absolutely relate to the Stats Widget frame width. Let's tackle this issue in a follow-up PR!

**Comment by ederrengifo** [comment]
@jsnmoon could we hold this change until we develop stage 1 of the widget upgrade?

File: apps/odyssey-stats/package.json
Comment (jsnmoon) [inline apps/odyssey-stats/package.json]: How did you add this dependency? It looks like the yarn lockfile wasn't updated and is causing build failures. I recommend using `yarn add some-dependency/here` to avoid this.

File: apps/odyssey-stats/package.json
Comment (jsnmoon) [inline apps/odyssey-stats/package.json:33]: I think we'll need to add `@automattic/calypso-color-schemes` as a dependency.

**Review by ederrengifo — COMMENTED** [review]
I think we need to adjust the line-height of the items list and be sure we follow these specs:
Screen Shot 2023-04-19 at 15 16 26

File: apps/odyssey-stats/src/widget/highlights.jsx
Comment (jsnmoon) [inline apps/odyssey-stats/src/widget/highlights.jsx]:
```suggestion
			title={ translate( 'View detailed stats for %(title)s', {
				args: {
					title: item.title
				},
				textOnly: true,
				comment: 'Text for anchor linking to a stats page for a given post/page',
			} ) }
```

No need to add a separate `aria-label` if a `title` is already provided.

**Review by jsnmoon — COMMENTED** [review]
image

Could we update the styling of the `No activity this period` notice to more closely resemble our design specifications? c6yabERlJE06tKLECfpnch-fi-6580-138157&t=kProk6bdXfHgnwac-4

**Comment by jsnmoon** [comment]
> @jsnmoon could we hold this change until we develop stage 1 of the widget upgrade? (@ederrengifo)

Could you clarify which change you're referring to? Was it the linked highlights?

**Comment by dognose24** [comment]
> Could we update the styling of the No activity this period notice to more closely resemble our design specifications?

@jsnmoon Nice catch! The empty state is fresh now.

截圖 2023-04-20 上午5 28 17

**Comment by kangzj** [comment]
I added a commit 145536a to make the font size smaller for axis labels, which fix some wrapping issue.

**Review by kangzj — APPROVED** [review]
LGTM

**Review by kangzj — APPROVED** [review]

**Review by kangzj — APPROVED** [review]
I can see all feedback are addressed. I'll go ahead and merge this, but if you have any more feedback, feel free to drop them here and we'll fix in follow up PRs. @jsnmoon @ederrengifo @keoshi

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7935173

Thank you @dognose24 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7935173).

**Pull request merged**
